src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/InputNPathComplexity.java

```diff
@@ -0,0 +1,168 @@
+package com.puppycrawl.tools.checkstyle.checks.metrics;
+// Advise: for lack of ambiguity try to make all factors prime numbers
+public class InputNPathComplexity {
+    //NP = 5
+    void testIfWithExpression() {
+        // NP = (if-range=1) + 1 + (expr=3) = 5
+        if (true && true || (true || true)) { }
+    }
+    
+    //NP = 5
+    void testIfElseWithExpression() {
+        // (if-range=1) + (else-range=1) + (expr=3) = 5
+        if (true && true || (true || true)) { }
+        else { }
+    }
+    
+    //NP = 4
+    int testSimpleSwitch() {
+        int a = 0;
+        // NP = (case-range(1)=1) + (case-range(1)=1) + (case-range(1)=1)
```

Comment:
1) please change "case-range(1)" to "case-range[1]" - lets use index syntax to avoid ambiguity. I updated xdoc file.
2) please fix index values in other cases
3) please apply 1)-2) to all Input files. If structure has strict names for parts then array indexes are not required. Array indexes are only for cases[i] and statement[i]